src/UQpy/surrogates/polynomial_chaos/physics_informed/ConstrainedPCE.py

```diff
@@ -0,0 +1,322 @@
+import numpy as np
+from scipy import special as sp
+from scipy.special import legendre
+from sklearn import linear_model as regresion
+from UQpy.surrogates import *
+from UQpy.distributions.collection import Uniform, Normal
+import UQpy.surrogates.polynomial_chaos.physics_informed.Utilities as utils
+import copy
+from beartype import beartype
+
+class ConstrainedPce:
+    @beartype
+    def __init__(self, pde_data, pde_pce, pce):
+        """
+        Class for construction of physics-informed PCE using Karush-Kuhn-Tucker normal equations
+
+        :param pde_data: an object of the :py:meth:`UQpy` :class:`PdeData` class
+        :param pde_pce: an object of the :py:meth:`UQpy` :class:`PdePce` class
+        :param pce:  initial pce, an object of the :py:meth:`UQpy` :class:`PolynomialChaosExpansion` class
+        """
+
+        self.pde_data = pde_data
+        self.pde_pce = pde_pce
+        self.initial_pce = pce
+
+        self.dirichlet_bc = self.pde_data.extract_dirichlet()
+        self.virtual_s = None
+        self.virtual_x = None
+        self.basis_extended = None
+        self.y_extended = None
+        self.kkt = None
+
+    def estimate_error(self, pce, verif_S):
+
+        ypce = pce.predict(pce.experimental_design_input)
+
+        err_data = (np.sum((pce.experimental_design_output - ypce) ** 2) / len(ypce))
+
+        err_pde = np.abs(
+            self.pde_pce.pde_eval(verif_S, pce, coefficients=pce.coefficients) - self.pde_pce.pderes_eval(verif_S,
+                                                                                                          multindex=pce.multi_index_set,
+                                                                                                          coefficients=pce.coefficients))
+        err_pde = np.mean(err_pde ** 2)
+        err_bc = self.pde_pce.bc_eval(len(verif_S), pce)
+        err_bc = np.mean(err_bc ** 2)
+        err_complete = (err_data + err_pde + err_bc)
+        return err_complete
+
+    def lar(self, n_PI=50, virtual_niters=False, max_niter=None, no_iter=False, minsize_basis=1, nvirtual=None,
```

Comment:
Is this one of the already implemented regression methods.? Is there is a need to duplicate it here.?

Reply:
No, this is a specific method only for physics-informed PCE. Although it use theory of Least Angle Regression, it adds additional equations describing physics in form of Karush-Kuhn-Tucker normal equations. I do not see any simple way how to use existing LAR instead of this new method.